uqcsbot/dominos_coupons.py

```diff
@@ -203,15 +223,18 @@ def _get_coupons(
 
     random.shuffle(coupons)
 
-    return coupons[:n]
+    return coupons[:n], failed_urls
 
 
-def _get_coupons_from_page(source: str) -> List[Coupon]:
+def _get_coupons_from_page(source: Literal["couponese", "frugalfeeds"]) -> List[Coupon]:
     """
     Strips results from html page and returns a list of Coupon(s)
     """
-    urls: List[str] = []
-    coupons: List[Coupon] = []
+
+    urls: Dict[str, str] = {
```

Comment:
I'd potentially make this dictionary a global constant, but that's up to you.

Reply:
Do you recommend a certain style? Is it fine to leave it as `urls` or should it be capitalized i.e. `URLS`? I'm aware constants should be fully capitalized but I'm unsure as it's a dictionary.

Reply:
All capitalised would be my thoughts, but I'm not certain.